pkg/webhook/topic_validator.go

```diff
@@ -175,9 +167,12 @@ func (s *webhookServer) checkExistingKafkaTopicCRs(ctx context.Context,
 		}
 
 		referredNamespace := kafkaTopic.Spec.ClusterRef.Namespace
-		if (kafkaTopic.GetNamespace() == clusterNamespace && referredNamespace == "") || referredNamespace == clusterNamespace {
-			foundKafkaTopic = &kafkaTopicList.Items[i]
-			break
+		referredName := kafkaTopic.Spec.ClusterRef.Name
+		if referredName == topic.Spec.ClusterRef.Name {
```

Comment:
This `if` statement will always evaluate to `true`

Reply:
Why? Its kafkaTopic.Spec.ClusterRef.Name == topic.Spec.ClusterRef.Name

Reply:
I see maybe we could change the name as it's quite easy to confuse.

Reply:
I think it can clearly see because the location of the referredName